plugins/autosuspend.koplugin/main.lua

```diff
@@ -11,10 +11,15 @@ local UIManager = require("ui/uimanager")
 local WidgetContainer = require("ui/widget/container/widgetcontainer")
 local logger = require("logger")
 local _ = require("gettext")
+local T = require("ffi/util").template
 
-local AutoSuspend = {
+local default_autoshutdown_timeout_seconds = 3*24*60*60
+
+local AutoSuspend = WidgetContainer:new{
+    name = "autosuspend",
+    is_doc_only = false,
+    autoshutdown_sec = G_reader_settings:readSetting("autoshutdown_timeout_seconds") or default_autoshutdown_timeout_seconds,
     settings = LuaSettings:open(DataStorage:getSettingsDir() .. "/koboautosuspend.lua"),
```

Comment:
You save into G_reader_settings. Is there still some reason to read this koboautosuspend.lua and loop thru candidates below? (alt: should you not save them into that plugin setting file, and forger about G_reader_settings here?)

Reply:
Now that you mention it, one of them is probably intended to be legacy. Given that the plugin-specific settings seem to stem from a migration out of the UIManager core, I'm guessing G_reader_settings is the one that should be deprecated.